hack/update-staging-client-go.sh

```diff
@@ -18,5 +18,57 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+V=""
+while getopts ":vs" opt; do
+  case $opt in
+    v) # increase verbosity
+      V="-v"
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      ;;
+  esac
+done
+readonly V
+
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
-"${KUBE_ROOT}"/staging/copy.sh
+cd ${KUBE_ROOT}
+
+# Create a temporary GOPATH for apimachinery and client-go, copy the current
+# HEAD into each and turn them into a git repo. Then we can run copy.sh with
+# this additional GOPATH. The Godeps.json will have invalid git SHA1s, but it's
+# good enough for running copy.sh. The actual k8s.io/apimachinery dependency
+# will be updated later by the github bot to point to the real k8s.io/apimachinery
+# github commit created by the bot just before."
+if [ "${USE_TEMP_DIR:-1}" = 1 ]; then
+    TEMP_STAGING_GOPATH=$(mktemp -d -t verify-staging-client-go.XXXXX)
+    echo "Creating the temporary staging GOPATH directory: ${TEMP_STAGING_GOPATH}"
+    cleanup() {
+        if [ "${KEEP_TEMP_DIR:-0}" != 1 ]; then
+            rm -rf "${TEMP_STAGING_GOPATH}"
+        fi
+    }
+    trap cleanup EXIT SIGINT
+    mkdir -p "${TEMP_STAGING_GOPATH}/src"
+    ln -s "${PWD}" "${TEMP_STAGING_GOPATH}/src/k8s.io"
+else
+    TEMP_STAGING_GOPATH="${GOPATH}"
+fi
+for PACKAGE in apimachinery client-go; do
```

Comment:
Yeah, I think you are right, I vaguely remember I had met the same problem before.